lib/schema/schema.js

```diff
@@ -24,8 +24,14 @@ var util = require('util');
 
 var misc = require('../misc.js');
 
+exports.validDeviceTime = function(val) {
+  if (!/^(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)$/.test(val)) {
```

Comment:
I know when we talked I agreed with this idea, but I now think this is not actually an improvement. This will pass dates like 2015-02-35 but fail for valid times with leap seconds (it's actually valid for seconds to have the value 60). It can only occur on 4 dates out of the year, but it does happen. On the other hand, the Date.parse code is documented to return NaN for invalid dates and does a much deeper job of validation. Please revert this change.